### DISCUSSION TITLE
Auth Custom Directive

### DISCUSSION THREAD

**Discussion by sbilello** [Netflix/dgs-framework#226] [General]
Hello!

Is there any documentation or support regarding adding custom directive in the schema:  [GraphQL Java Auth](https://www.graphql-java.com/documentation/v11/sdl-directives/ )
Maybe with codegen @srinivasankavitha?

Another approach would be to leverage spring security but I was wondering how to model it in the GraphQL schema.

Thanks

**Comment by srinivasankavitha** [comment]
You can define the directive in your schema as described in the graphql-java documentation. Then you can hook into the RuntimeWiring by using `@DgsRuntimeWiring`
```
We do have the ability to add to runtime wiring using @DgsRuntimeWiring annotation that you can add and use with your class that defines this custom directive. Something like this:

@DgsComponent
class CustomDirective implements SchemaDirectiveWiring {
...
}

@DgsRuntimeWiring
    fun addCustomDirective(builder: RuntimeWiring.Builder): RuntimeWiring.Builder {
        return builder.directive("custom", new CustomDirective())
    }
}
```
You can also refer to a related discussion here: https://github.com/Netflix/dgs-framework/discussions/155

If you don't need to wrap data fetchers etc. and just need to do some simple checks within the data fetcher, you also have access to it within the data fetcher from the DataFetchingEnvironment using `dfe.getFieldDefinition().getDirective("custom");` as described here: https://github.com/Netflix/dgs-framework/discussions/69

Hope that helps.

**Comment by ankit-joinwal** [comment]
There is now an example for custom directives in https://netflix.github.io/dgs/examples/

**Reply by taptew** [reply]
Is there an example of custom directive for the input fields?
